### PR TITLE
neovim 0.3.0

### PR DESCRIPTION
**Pull request by Meroje**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Review by ilovezfs — CHANGES_REQUESTED** [review]
```
Josephs-MacBook-Pro:~ joe$ curl -sL https://raw.githubusercontent.com/neovim/neovim/v0.3.0/third-party/cmake/BuildLuarocks.cmake|grep '${LUAROCKS_BUILDARGS}'
  ARGS build mpack 1.0.7-0 ${LUAROCKS_BUILDARGS}
  ARGS build lpeg 1.0.1-1 ${LUAROCKS_BUILDARGS}
  ARGS build inspect 3.1.1-0 ${LUAROCKS_BUILDARGS}
    ARGS build luabitop 1.0.2-3 ${LUAROCKS_BUILDARGS}
    ARGS build penlight 1.5.4-1 ${LUAROCKS_BUILDARGS}
    ARGS build busted 2.0.rc12-1 ${LUAROCKS_BUILDARGS}
    ARGS build luacheck 0.21.2-1 ${LUAROCKS_BUILDARGS}
    ARGS make ${LUAROCKS_BUILDARGS} ${LUV_ARGS}
    ARGS build nvim-client 0.1.0-1 ${LUAROCKS_BUILDARGS}
```
It looks like there's an upgrade for mpack and some new luarocks needed.

CC @javian @DomT4

**Comment by Meroje** [comment]
Just had a quick look, luabitop penlight busted and nvim-client do not have src rock. Not sure how those work.

**Comment by Meroje** [comment]
I tried bumping to lua 5.3 but luabitop did not build
```
luarocks-5.3
build
build/src/luabitop/luabitop-1.0.2-1.src.rock
--tree=.

bit.c:83:2: error: "Unknown number type, check LUA_NUMBER_* in luaconf.h"
#error "Unknown number type, check LUA_NUMBER_* in luaconf.h"
 ^
1 error generated.

Error: Build error: Failed compiling object bit.o
env MACOSX_DEPLOYMENT_TARGET=10.8 gcc -O2 -fPIC -I/usr/local/include/lua5.3 -c bit.c -o bit.o
```

File: Formula/neovim.rb [hunk 1]
```diff
@@ -30,8 +29,88 @@ class Neovim < Formula
...
+    sha256 "d1c706d48efc7279d33f5ea123acb4d27e2ee93e364bedbe07f2c9c8d0ad3d24"
+  end
+
+  resource "dkjson" do
+    url "https://luarocks.org/dkjson-2.5-2.src.rock", :using => :nounzip
+    sha256 "0391ebe73cfcee7d6b3d5dd5098e185c2103118e644688484beea665f15fc9e3"
+  end
+
+  resource "say" do
+    url "https://github.com/Olivine-Labs/say/archive/v1.3-1.tar.gz"
```
Comment (ilovezfs) [inline Formula/neovim.rb]:
```
version "1.3-1"
```

File: Formula/neovim.rb [hunk 2]
```diff
@@ -30,8 +29,88 @@ class Neovim < Formula
...
+    sha256 "23e8cd378bb4ab1693279100a785acb2246418e3570b7de7d995b5847b3507ca"
+  end
+
+  resource "luassert" do
+    url "https://github.com/Olivine-Labs/luassert/archive/v1.7.10.tar.gz"
+    sha256 "f9f8347727c2a4aa8af30d88a0de0314f04cd681b60430e24f6ec0ed393e12e1"
+  end
+
+  resource "mediator_lua" do
+    url "https://github.com/Olivine-Labs/mediator_lua/archive/v1.1.2-0.tar.gz"
```
Comment (ilovezfs) [inline Formula/neovim.rb]:
```
version "1.1.2-0"
```

File: Formula/neovim.rb [hunk 3]
```diff
@@ -30,8 +29,88 @@ class Neovim < Formula
...
+    sha256 "faf5859fd2081be4e9e4fb8873a2dc65f7eff3fd93d6dd14da65f8e123fcff9b"
+  end
+
+  resource "busted" do
+    url "https://github.com/Olivine-Labs/busted/archive/v2.0.rc12-1.tar.gz"
```
Comment (ilovezfs) [inline Formula/neovim.rb]:
```
version "2.0.rc12-1"
```

File: Formula/neovim.rb [hunk 4]
```diff
@@ -30,8 +29,88 @@ class Neovim < Formula
...
+    sha256 "c44286468babcc38e90f036d25471ab92f19a8a0a68482e0c45a9cfaeb1c0e35"
+  end
+
+  resource "luacheck" do
+    url "https://luarocks.org/manifests/mpeterv/luacheck-0.21.2-1.src.rock", :using => :nounzip
```
Comment (ilovezfs) [inline Formula/neovim.rb]:
```
version "0.21.2-1"
```

File: Formula/neovim.rb [hunk 5]
```diff
@@ -30,8 +29,88 @@ class Neovim < Formula
...
+    sha256 "c9e9b3bf1610e382043c6348417864541327108da92290a3be454c40be439953"
+  end
+
+  resource "luv" do
+    url "https://luarocks.org/luv-1.9.1-1.src.rock", :using => :nounzip
+    sha256 "d72db8321d8b3be925e1c14e6c13081466d1c31420f600154ab5c77fe6974fac"
+  end
+
+  resource "coxpcall" do
+    url "https://luarocks.org/coxpcall-1.17.0-1.src.rock", :using => :nounzip
```
Comment (ilovezfs) [inline Formula/neovim.rb]:
```
version "1.17.0-1"
```

File: Formula/neovim.rb [hunk 6]
```diff
@@ -30,8 +29,88 @@ class Neovim < Formula
...
+    sha256 "11feb07f08927c39b0b93e8c0bbaf15433f86155cba4820a31a09f4754ab3258"
+  end
+
+  resource "nvim-client" do
+    url "https://github.com/neovim/lua-client/archive/0.1.0-1.tar.gz"
```
Comment (ilovezfs) [inline Formula/neovim.rb]:
```
version "0.1.0-1"
```

File: Formula/neovim.rb [hunk 7]
```diff
@@ -30,8 +29,94 @@ class Neovim < Formula
...
+    sha256 "d1c706d48efc7279d33f5ea123acb4d27e2ee93e364bedbe07f2c9c8d0ad3d24"
+  end
+
+  resource "dkjson" do
+    url "https://luarocks.org/dkjson-2.5-2.src.rock", :using => :nounzip
+    sha256 "0391ebe73cfcee7d6b3d5dd5098e185c2103118e644688484beea665f15fc9e3"
+  end
+
+  resource "say" do
+    version "1.3-1"
```
Comment (ilovezfs) [inline Formula/neovim.rb]: goes under url

Comment (Meroje) [reply Formula/neovim.rb]: right, sorry about that

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks so much for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock, @Meroje!

**Comment by ilovezfs** [comment]
Closed via https://github.com/Homebrew/homebrew-core/commit/b2a970223116aff26c579ecbece2a256d43b7f5d

**Pull request closed without merging**